app/views/idv/in_person/state_id.html.erb

```diff
@@ -16,6 +19,7 @@
     <%= render ValidatedFieldComponent.new(
           name: :first_name,
           form: f,
+          input_html: { value: pii[:first_name] },
```

Comment:
We could avoid assigning the value explicitly if we initialize SimpleForm with a form object that's populated with the values. Maybe something roughly like:

```rb
def extra_view_variables
  {
    form: Idv::StateIdForm.new(flow_session[:pii_from_user]),
    # ...
  }
end
```

```erb
<%= validated_form_for form do |f| %>
```

It might get a little tricky with the input names though, since FlowStateMachine assumes to [pick form values from the flow's name](https://github.com/18F/identity-idp/blob/7bfdd3b7ddd56539eddd009959e48c857f8b9108/app/services/flow/base_step.rb#L61-L63) (`:doc_auth` in this case). We could control that by assigning the form's `model_name` as 'DocAuth' in all cases, but it feels a little hacky. 🤷 

https://github.com/18F/identity-idp/blob/7bfdd3b7ddd56539eddd009959e48c857f8b9108/app/forms/idv/state_id_form.rb#L10-L12

Reply:
Oof, yeah, it feels pretty hacky to me with having to name all of the forms 'DocAuth'. How about I add a ticket to look into this post-pilot? I think there'll be some refactoring work to fully pull the doc auth and IPP flows apart and this could be an enhancement that allows

What's the general attitude towards `WILLFIX` comments left in the code? I'm thinking I could leave such comments linking to the enhancement ticket but they might be around for quite a few weeks depending on when we can get to it

Reply:
@sheldon-b Yeah, I wouldn't consider this a blocker, but I think it is highlighting a couple specific issues:

- That we're extending DocAuth base classes in the non-DocAuth flow
- That FlowStateMachine is looking for a specific, constant form key
   - This is related to the first bullet point because it's from [how the base step is initialized](https://github.com/18F/identity-idp/blob/ec17ed1f7ec2ae0536268e716823d86a2ae3515e/app/services/idv/steps/doc_auth_base_step.rb#L4-L6)
   - However, it might be nice if [`flow_params`](https://github.com/18F/identity-idp/blob/ec17ed1f7ec2ae0536268e716823d86a2ae3515e/app/services/flow/base_step.rb#L61-L63) could reference [the relevant form](https://github.com/18F/identity-idp/blob/ec17ed1f7ec2ae0536268e716823d86a2ae3515e/app/services/flow/base_step.rb#L50-L52) in a way that it could pull parameters from `form.class.model_name` instead

Later refactoring may be fine, though I'd wonder if it would eventually be made redundant as [FlowStateMachine v2](https://cm-jira.usa.gov/browse/LG-4159) would seek to reimplement these steps anyways.

To the question about WILLFIX: You may be able to add it, but generally these kinds of comments are [prevented by our CodeClimate configuration](https://github.com/18F/identity-idp/blob/ec17ed1f7ec2ae0536268e716823d86a2ae3515e/.codeclimate.yml#L43-L53).

Reply:
Yeah that makes sense. I'll create a ticket to track this tech debt, I would hope that we can prioritize it sometime after the pilot